### ISSUE TITLE
ublock keeps generating/adding blocked items on southpark.com, a 1000 every 8 minutes

### ISSUE THREAD

**Issue by hcbd** [gorhill/uBlock#2353]
### Describe the issue

The blocked element counter keeps going up when watching a full episode on southpark.com
It seems the blocking of ad's on this website makes the website produce new links/objects on the page when the orginal link/object isn't loaded. This makes ublock consume RAM, slow's down your browser and eventually lock it up. It generates about a 1000 blocked elements within 8 minutes.
I would call it a DDOS attack on resources, but don't know if it's done on purpose.

### One or more specific URLs where the issue occurs

This happens when viewing any full episode on southpark.cc.com.
an example:
http://southpark.cc.com/full-episodes/s20e02-skank-hunt

### Steps for anyone to reproduce the issue

Reproduce:
1. visit www.southpark.com
2. go to the full episodes part of the website
3. pick any episode and hit play.
4. watch the blocked elements counter go up like crazy (number on the ublock icon next to the urlbar)
5. mine reached 5000 blocked elements after watching 1 episode.

### Your settings

Default settings that come with the installation from addons.mozilla.org

- Browser/version: Firefox 51.0.1 (32-bit) Windows 10 enterprise (in a VM)
- uBlock Origin version: 1.10.6

##### Your filter lists

Default filter lists - no extra filters selected

##### Your custom filters (if any)

No custom filters

**Comment by gorhill** [comment]
> This makes ublock consume RAM

You have _actual_ evidence of this? The block count is **completely unrelated** to the RAM usage of uBO.

**Comment by mapx-** [comment]
Could be related to https://bugzilla.mozilla.org/show_bug.cgi?id=1326095 ?

**Comment by gorhill** [comment]
See <https://bugzilla.mozilla.org/show_bug.cgi?id=1326095#c29>.

**Comment by mapx-** [comment]
right..

**Comment by hcbd** [comment]
Not sure if this is only a filter issue, because the RAM does grow and Firefox does become unresponsive after a while. I do not see this behaviour on other (simular) websites, with the same system.
but i will check out the firefox bug and if a nightbuild doesn't help post a video/screenshots here.

**Comment by gorhill** [comment]
Also use the logger to find out what is blocked repeatedly.

**Comment by gorhill** [comment]
Also, using a US VPN, I can't watch anything, I get an error message about sponsors not being able to show their stuff.

**Comment by hcbd** [comment]
The logger doesn't show anything being blocked.
And i'm visiting the website from the Netherlands (no vpn)

**Comment by hcbd** [comment]
screenshot at start of video: http://nl.tinypic.com/r/244b577/9
screenshot after 6 minutes of video: http://nl.tinypic.com/r/v5fkud/9
firefox doesn't seem to use a Lot of RAM but does become slower/unresponsive

**Comment by gorhill** [comment]
Well I can't even play the video without any blocker, allowing all 3rd-party cookies and Flash, etc.

**Comment by gorhill** [comment]
> The logger doesn't show anything being blocked

Select the _All_ entry in the drop-down list at the top.

**Comment by hcbd** [comment]
Selected the All from the dropdown list...
still no entries: http://nl.tinypic.com/r/j8eadc/9

**Comment by gorhill** [comment]
> still no entries

Just in case, force a refresh of the logger page (F5).

**Comment by gorhill** [comment]
You could also try to close the `much.com` tab. Maybe the amount of request fired by the page make it difficult for the logger to refresh. When you close the tab, the logger may finally be given enough time to report the content, though the entries will appear dimmed. At least you will know what is being blocked.

**Comment by hcbd** [comment]
found out why it doesn't work:
needed to remove the # sign from the url
now it show the following: http://nl.tinypic.com/r/2qltzba/9

**Comment by gorhill** [comment]
That is an EasyPrivacy filter. Try this exception filter:

    @@/b/ss/*&aqe=$domain=southpark.cc.com

**Comment by hcbd** [comment]
added it to my filters, no effect

**Comment by gorhill** [comment]
"no effect" as in...? The network requests are still blocked? The same network request? I have to second guess a lot of stuff here. Try:

    @@/b/ss/*&aqe=$object,domain=cc.com

If this does not work, try to create an exception filter yourself using the [wizard in the logger](https://github.com/gorhill/uBlock/wiki/The-logger#static-network-filters).

**Comment by hcbd** [comment]
No change in behaviour in the logger (reloaded the page). the 'object' is still being blocked.

the logger makes the following filter:
@@||sc.cc.com/b/ss$object,domain=southpark.cc.com

**Comment by gorhill** [comment]
> the 'object' is still being blocked

By the same filter as seen in <http://nl.tinypic.com/r/2qltzba/9>?

**Comment by hcbd** [comment]
yes it is/was.

Thanks for the help 👍 
the filter now works, there is another one, but will check it out (it doesn't reload every second)

i will close this issue

**Comment by gorhill** [comment]
So this is a filter issue, and given that the filter is in EasyPrivacy, this will affect all blockers using EasyPrivacy. This should be reported to EasyPrivacy maintainers.

**Comment by hcbd** [comment]
I will report it to the easyprivacy maintainers

**Comment by gorhill** [comment]
@mapx- 

> right..

Not sure what to make of this comment.

In any case, it made me worry that maybe I haven't been thorough enough with my own tests, so I decided to repeat a couple of times again to be sure, except that this time I disabled all filter lists in uBO and added a single filter `*$xmlhttprequest` to behave just like the minimalist extension used to reproduce the issue.

My initial tests are confirmed: the leak still does not occur with uBO, but systematically occur with the minimalist extension by just opening one Youtube video tab. This reinforced that the issue is with a code path visited when using `nsIContentPolicy`, and not with a code path visited when using an HTTP observer.
